frontend/src/pages/facemash.js

```diff
@@ -30,74 +33,79 @@ class FaceMash extends Component<Props> {
   constructor(props) {
     super(props)
     this.state = {
-      results: this.props.result,
       error: false,
       message: ''
     }
   }
 
-  static async getInitialProps({ query }) {
-    // check whether query.id is real candidate
-    // TODO: do not fetch again, if user hasn't responded to previous match
-    try {
-      const { result } = await getCandidateMatch()
-      if (result === undefined) {
-        return { error: 'Bad Request' }
-      }
-      return { result }
-    } catch (err) {
-      console.log('Candidate Page error: ', err.message)
-      return { error: 'Bad Request' }
+  async getNewMatch() {
+    const { result } = await getCandidateMatch()
+    const { generateMatchData } = this.props
+    if (result === undefined) {
+      console.log('Could not get new FaceMash match')
+      generateMatchData(null, null, null)
+    } else {
+      generateMatchData(result.candidate1, result.candidate2, result.matchID)
+    }
+  }
+
+  componentDidMount() {
+    const { candidates } = this.props
+    if (candidates == null || candidates.length != 2) {
+      this.getNewMatch()
     }
   }
 
   handleClick = async e => {
     const winner = e.target.name
     try {
-      const { candidate1, candidate2, matchID } = this.state.results
+      const { candidates, matchID } = this.props
       const res = await setMatchWinner(
-        candidate1._id,
-        candidate2._id,
-        winner === '0' ? candidate1._id : candidate2._id,
+        candidates[0]._id,
+        candidates[1]._id,
+        winner === '0' ? candidates[0]._id : candidates[1]._id,
         matchID
       )
       if (res.success) {
         this.setState({
           success: 'Successfully Submitted'
         })
       }
```

Comment:
I know you didn't change the backend endpoint for this but:
- the backend endpoint shouldn't return `{ success: true }`, instead it should return something like this: 
``` Javascript
{
  "code": 200, 
  "message": "", 
  "result": {}, 
  "success": true
}
```
I'd also add an else statement if the endpoint returns success `false`, probably just `console.log`and make an alert pop up.